reframe/utility/jsonext.py

```diff
@@ -40,14 +41,35 @@ def encode(obj):
     return None
 
 
+def _make_json_friendly(obj):
```

Comment:
I would rather call this something that shows that it's for the keys, because the `encode()` above is meant to make the JSON values at least friendly. The problem as I see it is that `json.dump` does not delegate to the `default` callable when it doesn't know how to convert keys and issues an error.